dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventQueryParams.java

```diff
@@ -450,6 +459,17 @@ public Map<TrackedEntityAttribute, List<QueryFilter>> getAttributes() {
     return this.attributes;
   }
 
+  /** Returns attributes that are only ordered by and not present in any filter. */
+  public Set<TrackedEntityAttribute> leftJoinAttributes() {
+    return SetUtils.difference(getOrderAttributes(), filterableAttributes().keySet());
+  }
+
+  public Map<TrackedEntityAttribute, List<QueryFilter>> filterableAttributes() {
+    return attributes.entrySet().stream()
+        .filter(a -> !a.getValue().isEmpty())
```

Comment:
If we filter by an attribute and we do not add any filter on the value this should still translate to a `INNER JOIN`.
Do we have a test like `shouldFilterEventsByAttributeWithNoValuesAndFilterOutEventsWithATrackedEntityWithoutThatAttribute`